redwood/src/lib.rs

```diff
@@ -0,0 +1,233 @@
+        .set_cipher_suite(CipherSuite::RSA4k)
+        .add_userid(format!("Source Key <{}>", email))
+        .set_creation_time(
+            // All reply keypairs will be "created" on the same day SecureDrop (then
+            // Strongbox) was publicly released for the first time.
+            // https://www.newyorker.com/news/news-desk/strongbox-and-aaron-swartz
+            SystemTime::UNIX_EPOCH
+                // Equivalent to 2013-05-14
+                .checked_add(Duration::from_secs(1368507600))
+                // unwrap: Safe because the value is fixed and we know it won't overflow
+                .unwrap(),
+        )
+        .add_storage_encryption_subkey()
+        .set_password(Some(passphrase.into()))
+        .generate()?;
+    let public_key = String::from_utf8(cert.armored().to_vec()?)?;
+    let secret_key = String::from_utf8(cert.as_tsk().armored().to_vec()?)?;
+    Ok((public_key, secret_key, format!("{}", cert.fingerprint())))
+}
+
+/// Encrypt a message (text) for the specified recipients. The list of
+/// recipients is a set of PGP public keys. The encrypted message will
+/// be written to `destination`.
+#[pyfunction]
+pub fn encrypt_message(
+    recipients: Vec<String>,
+    plaintext: String,
+    destination: PathBuf,
+) -> Result<()> {
+    let plaintext = plaintext.as_bytes();
+    encrypt(&recipients, plaintext, &destination)
+}
+
+/// Encrypt a file that's already on disk for the specified recipients.
+/// The list of recipients is a set of PGP public keys. The encrypted file
+/// will be written to `destination`.
+#[pyfunction]
+pub fn encrypt_file(
+    recipients: Vec<String>,
+    plaintext: PathBuf,
+    destination: PathBuf,
+) -> Result<()> {
+    let plaintext = File::open(plaintext)?;
+    encrypt(&recipients, plaintext, &destination)
+}
+
+/// Helper function to encrypt readable things.
+fn encrypt(
+    recipients: &[String],
+    mut plaintext: impl Read,
+    destination: &Path,
+) -> Result<()> {
+    let p = &StandardPolicy::new();
+    let mut certs = vec![];
+    let mut recipient_keys = vec![];
+    for recipient in recipients {
+        certs.push(Cert::from_str(recipient)?);
+    }
+    for cert in certs.iter() {
+        for key in cert
+            .keys()
+            .with_policy(p, None)
+            .supported()
+            .alive()
+            .revoked(false)
+        {
+            recipient_keys.push(key);
+        }
+    }
+
+    let mut sink = File::create(destination)?;
+    let message = Message::new(&mut sink);
+    let message = Armorer::new(message).build()?;
+    let message = Encryptor::for_recipients(message, recipient_keys).build()?;
+    let mut message = LiteralWriter::new(message).build()?;
+
+    io::copy(&mut plaintext, &mut message)?;
+
+    message.finalize()?;
+
+    Ok(())
+}
+
+/// Decrypt the given ciphertext using the specified private key that is locked
+/// using the provided passphrase. It is assumed that the plaintext is UTF-8.
```

Comment:
Nit for clarity: Are "given", "specified", and "provided" interchangeable in this comment?  If so, I would suggest something like:

```suggestion
/// Given a ciphertext, private key, and passphrase, unlock the private key with
/// the passphrase, and use it to decrypt the ciphertext.  It is assumed that the
/// plaintext is UTF-8.
```

Reply:
Yes, ty, applied.